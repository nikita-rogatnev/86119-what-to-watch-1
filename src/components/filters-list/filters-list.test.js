import React from 'react';
import renderer from 'react-test-renderer';

import FiltersList from './filters-list';

import moviesList from '../../mocks/movies';

it(`FiltersList renders correctly`, () => {
  const activeGenre = `All genres`;

  let filtersList = Array.from(new Set(moviesList.map(({genre}) => genre)));
  filtersList.unshift(`All genres`);

  const tree = renderer.create(<FiltersList
    movies={moviesList}
    allGenres={filtersList}
    activeItem={activeGenre}
    onActiveGenreChange={() => {
    }}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
