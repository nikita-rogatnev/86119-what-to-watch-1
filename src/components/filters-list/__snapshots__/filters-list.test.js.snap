// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersList renders correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Adventure
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Action
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Crime
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Drama
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Fantasy
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Thriller
    </a>
  </li>
</ul>
`;
